src/tools/data.js

```diff
@@ -142,14 +142,17 @@ const recurseExtract = (obj, matcher) => {
 const _IGNORE = {};
 
 // Flatten a nested object.
-const flattenPaths = (data, sep = '.') => {
+const flattenPaths = (data, { preserve = {} } = {}) => {
   const out = {};
-  const recurse = (obj, prop) => {
-    prop = prop || '';
-    if (isPlainObj(obj)) {
-      Object.keys(obj).map(key => {
-        const value = obj[key];
-        const newProp = prop ? prop + sep + key : key;
+  const recurse = (obj, prop = '') => {
+    if (_.isPlainObject(obj)) {
+      Object.entries(obj).forEach(([key, value]) => {
+        const newProp = prop ? `${prop}.${key}` : key;
+
+        if (preserve[prop]) {
+          out[newProp] = value;
```

Comment:
We _could_ hit eject here and `return` from the loop. But this could be potentially breaking. For now, this change is additive to what we have today.

Reply:
if you think you want to break previous behavior, today is a great day for it. last breaking change for at least a little bit

Reply:
I'm fine with keeping the current behavior. Doesn't seem like there's any reason to break for this.